synapse/handlers/user_directory.py

```diff
@@ -349,21 +346,32 @@ def _handle_deltas(self, deltas):
                         # need to remove those users or not
                         user_ids = yield self.store.get_users_in_dir_due_to_room(room_id)
                         for user_id in user_ids:
-                            yield self._handle_remove_user(room_id, user_id)
+                            is_support = yield self.store.is_support_user(state_key)
```

Comment:
should this be `s/state_key/user_id/`? But surely the support user would never appear in the results from `get_users_in_dir_due_to_room` so I'm confused.

Reply:
should this be s/state_key/user_id/? err yes you are right - really hard to write a test for this method.

But surely the support user would never appear in the results from get_users_in_dir_due_to_room so I'm confused - as per above is only there out of defence - I'd be happy remove and save on the added complexity.